src/Runner.Worker/StepsRunner.cs

```diff
@@ -28,6 +29,27 @@ public interface IStep
         Task RunAsync();
     }
 
+    public static class IStepExtension 
+    {
+        public static void WriteWebhookPayload(this IStep step, IHostContext hostContext, Tracing trace) 
```

Comment:
Maybe just move this to `ExecutionContext.cs` so it would be `ExecutionContext.CreateGitHubEventPayloadFile()`?